### PR TITLE
Pyinvoke CLI and 2-step deployment

### PR DESCRIPTION
**Pull request by rdettai**
To facilitate the integration with the CI tooling, it is better to avoid using the local-exec provider in Terraform. To achieve that we need to:
- split the terraform deployment into 2 steps, one that deploys the repository to ECR and one that deploys the rest
- have a more flexible scripting tool for the CLI, we have chosen Python and PyInvoke.

### :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

- This moves the Terraform files around to have the 2 step deployment. I am open to a better naming than `step-1` and `step-2`
- Rewrite the CLI in python:
  - the `tasks.py` file is a PyInvoke task module with the same features as the makefile
  - to reduce the dependencies, I have added a Docker executor that can be bootstraped from the `vast-cloud` script
  - **changed default.env to .env** to have a more standard configuration setup
  - lost auto-completion feature

### CODE REVIEW

**Comment by rdettai** [comment]
Todo: add back git describe to get latest release version

**Comment by rdettai** [comment]
TODO: bash auto completion + destroy fails when the lamda-image and step-2 are not deployed

**Review by dispanser — APPROVED** [review]
I'm mostly happy with this! It really seems easier to grok, even though the new python-based thing is considerably larger than the `Makefile` it replaces.

As for naming the deployment steps, I'd like to see something that captures the intent of each step, maybe:
- `deploy-container-registry`
- `deploy-vast-containers`

The names aren't great either, but they allow for improving granularity later.

File: cloud/aws/vast-cloud [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    || docker build \
+        -t tenzir/vast-cloud-cli \
+        -f docker/cli.Dockerfile \
+        --build-arg UID=$(id -u) \
+        --build-arg GID=$(id -g) \
+        --build-arg DOCKER_ID=$(stat -c '%g' ${DOCKER_SOCKET}) \
+        .
+
+docker run -it \
+    --mount type=bind,source=${DOCKER_SOCKET},target=/var/run/docker.sock \
```
[the remark this answers is not in the document]
Comment (rdettai) [reply cloud/aws/vast-cloud:21]: The docker daemon is not running inside docker but on the host. [This post](https://jpetazzo.github.io/2015/09/03/do-not-use-docker-in-docker-for-ci/) is a bit lengthy but explains the ins and outs of docker in docker 🙂

File: cloud/aws/step-2/main.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    tags = {
+      module      = module.env.module_name
+      provisioner = "terraform"
+      stage       = terraform.workspace
+    }
+  }
+}
+
+# This provider will create enpoints in the existing VPC to allow the metric collection.
+# It can be reconfigured to use other credentials to setup cross-account monitoring.
```
[the remark this answers is not in the document]
Comment (rdettai) [reply cloud/aws/step-2/main.tf:26]: The configuration is the same for now, but I wanted to make it ready to support more generic configurations, such as cross-region or cross-account VPC peerings. I usually try to apply YAGNI to this kind of forward thinking weirdness, but I don't know why here I really feel it is important 😅 .

File: cloud/aws/README.md [hunk 3]
```diff
@@ -16,122 +16,119 @@ throughput-latency trade-off.
...
-![AWS Architecture](https://user-images.githubusercontent.com/53797/157068659-41d7c9fe-8403-40d0-9cdd-dae66f0bf62e.png)
+![AWS
+Architecture](https://user-images.githubusercontent.com/53797/157068659-41d7c9fe-8403-40d0-9cdd-dae66f0bf62e.png)
 
 ### Usage
 
-The provided [Makefile](Makefile) wraps [Terraform](https://www.terraform.io/)
-and `aws` CLI commands for a streamlined UX. You can list the available higher
-level commands with `make help`.
+The provided CLI using [PyInvoke tasks module](tasks) wraps
```
[the remark this answers is not in the document]
Comment (rdettai) [reply cloud/aws/README.md]: sorry for that

**Pull request merged**